test/base_class.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+import unittest
+from test.conftest import get_test_config
+
+
+class BaseTestClass(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls: unittest.TestCase) -> None:
+        props = get_test_config()
+        for prop in [
+            "cfg",
+            "ctx",
+            "serviceImpl",
+            "wsClient",
+            "wsName",
+            "au",
+            "dfu",
+            "foft",
+        ]:
+            setattr(cls, prop, props[prop])
+
+        cls.prepare_data()
+
+    @classmethod
+    def prepare_data(cls: unittest.TestCase) -> None:
+        """Set up test fixtures.
+        Subclasses should implement whatever functions are required to set up the test cases.
+
+        :param cls: class object
+        :type cls: BaseTestClass
+        """
```

Comment:
I forget the PEP, but I think it's good practice to drop a `pass` in here for a function to be defined by children. Possibly also `@abstractmethod` ? Though this isn't technically an abstract base class...